lib/tasks/data/export.rake

```diff
@@ -6,32 +6,51 @@ namespace :data do
 
       admin = User.find 1
       scores_model_by_id = {
-        56 => Course56ProjectScore,
-        57 => Course57ProjectScore,
-        58 => Course58ProjectScore,
         71 => Course71ProjectScore,
         73 => Course73ProjectScore
       }
       course_id = args[:course_id].to_i
       klass = scores_model_by_id[course_id]
+      klass.column_names.delete('id')
 
-      puts("Exporting data for Course #{course_id}")
+      puts("### Exporting data for Course #{course_id} - #{Time.now.strftime("%Y-%m-%d %H:%M:%S %Z")}")
       output = Export::GradeDownload.csv(admin, course_id: course_id)
 
       # we want to truncate and not just delete so that the ids start back at 1
       # the way Periscope syncing works means that old rows will get overwritten this
       # way so that we don't waste space
       ActiveRecord::Base.connection.execute("TRUNCATE #{klass.table_name}")
-      puts("Transferring to database table")
-      skip_row = true # skip the header
+      puts("  Transferring to database table")
+      header_row = true # skip the header
       CSV.parse(output) do |row|
-        unless skip_row then
-          colnames_minus_id = klass.column_names.delete('id')
-          vals_by_colname = Hash[klass.column_names.zip row]
+        if header_row then
+          header_row = false
+          # Validate that the grade export format matches the table we created to hold it.
+          # It can change if we add new assignments, add or delete rubric rows, or edit rubric criteria
+          # Also, the order of the columns matters.
+          # See, for example, db/migrate/20191016032556_replace_project_scores73_table.rb for how the
+          # table column names where created.
+          if row.length != klass.column_names.length then
+            raise RuntimeError.new("The number of columns in Export::GradeDownload (#{row.length}) doesn't match the number of columns in #{klass.name} (#{klass.column_names.length}).")
+          end
+          row.each_with_index do |parsed_colname, i| 
+            # Don't do an exact match bc we may have had to deal with weird chars when creating the table.
+            # Also, the table column name may be truncated (postgres has a max column name lenght)
+            header_colname_stripped = parsed_colname.downcase.gsub(/[^0-9a-z]/, '')
+            klass_colname_stripped = klass.column_names[i].downcase.gsub(/[^0-9a-z]/, '')
+            unless klass_colname_stripped.start_with? header_colname_stripped[0..klass_colname_stripped.length-1]
```

Comment:
if these are both the same length, do we need start_with or just ==?

Reply:
The `header_colname` could become smaller if the rubric or assignment name is shortened. We still want it to match though. We just can't use start_with if the `header_colname` is longer b/c that returns false.